flashinit fail with RBOOT_ENABLED = 1

hi,

I'm trying to use flashinit with my project but it fail when rboot is enabled.
I'm using v3.1.1 Hotfix and development (11a3912), both generate invalid command to esptool.

testing Sming/samples/Basic_rBoot:

$ make flashinit
Flash init data default and blank data.
/opt/esp-open-sdk/esptool/esptool.py -p /dev/ttyUSB0 -b 115200 erase_flash
esptool.py v1.2
Connecting...
Running Cesanta flasher stub...
Erasing flash (this may take a while)...
Erase took 9.9 seconds
/opt/esp-open-sdk/esptool/esptool.py -p /dev/ttyUSB0 -b 115200 write_flash -ff 40m -fm qio -fs 32m 0x3fc000 /opt/esp-open-sdk/sdk/bin/esp_init_data_default.bin 0x3fe000 /opt/esp-open-sdk/sdk/bin/blank.bin  /opt/Sming/Sming/compiler/data/blankfs.bin
usage: esptool write_flash [-h] [--flash_freq {40m,26m,20m,80m}]
                           [--flash_mode {qio,qout,dio,dout}]
                           [--flash_size {detect,4m,2m,8m,16m,32m,16m-c1,32m-c1,32m-c2}]
                           [--no-progress] [--verify]
                           <address> <filename> [<address> <filename> ...]
esptool write_flash: error: argument <address> <filename>: Address "/opt/Sming/Sming/compiler/data/blankfs.bin" must be a number
/opt/Sming/Sming/Makefile-rboot.mk:525: recipe for target 'flashinit' failed
make: *** [flashinit] Error 2

--- a/Sming/Makefile-rboot.mk
+++ b/Sming/Makefile-rboot.mk
@@ -152,6 +152,11 @@ SPIFF_FILES ?= files
 BUILD_BASE	= out/build
 FW_BASE		= out/firmware
 
+#Firmware memory layout info files
+FW_MEMINFO_NEW = $(FW_BASE)/fwMeminfo.new
+FW_MEMINFO_OLD = $(FW_BASE)/fwMeminfo.old
+FW_MEMINFO_SAVED = out/fwMeminfo
+
 RBOOT_ROM_0  := $(addprefix $(FW_BASE)/,$(RBOOT_ROM_0).bin)
 RBOOT_ROM_1  := $(addprefix $(FW_BASE)/,$(RBOOT_ROM_1).bin)
 
@@ -445,6 +450,14 @@ $(TARGET_OUT_0): $(APP_AR)
 	$(vecho) "LD $@"
 	$(Q) $(LD) -L$(USER_LIBDIR) -L$(SDK_LIBDIR) -L$(BUILD_BASE) $(RBOOT_LD_0) $(LDFLAGS) -Wl,--start-group $(APP_AR) $(LIBS) -Wl,--end-group -o $@
 	$(Q) $(STRIP) $@
+	
+	$(Q) $(MEMANALYZER) $@ > $(FW_MEMINFO_NEW)
+	
+	$(Q) if [ -f "$(FW_MEMINFO_NEW)" -a -f "$(FW_MEMINFO_OLD)" ]; then \
+	  awk -F "|" 'FILENAME == "$(FW_MEMINFO_OLD)" { arr[$$1]=$$5 } FILENAME == "$(FW_MEMINFO_NEW)" { if (arr[$$1] != $$5){printf "%s%s%+d%s", substr($$0, 1, length($$0) - 1)," (",$$5 - arr[$$1],")\n" } else {print $$0} }' $(FW_MEMINFO_OLD) $(FW_MEMINFO_NEW); \
+	elif [ -f "$(FW_MEMINFO_NEW)" ]; then \
+	  cat $(FW_MEMINFO_NEW); \
+	fi
 
 
 $(TARGET_OUT_1): $(APP_AR)
@@ -524,7 +537,7 @@ terminal:
 flashinit:
 	$(vecho) "Flash init data default and blank data."
 	$(ESPTOOL) -p $(COM_PORT) -b $(COM_SPEED_ESPTOOL) erase_flash
-	$(ESPTOOL) -p $(COM_PORT) -b $(COM_SPEED_ESPTOOL) write_flash $(flashimageoptions) $(INIT_BIN_ADDR) $(SDK_BASE)/bin/esp_init_data_default.bin $(BLANK_BIN_ADDR) $(SDK_BASE)/bin/blank.bin $(SPIFF_START_OFFSET) $(SMING_HOME)/compiler/data/blankfs.bin
+	$(ESPTOOL) -p $(COM_PORT) -b $(COM_SPEED_ESPTOOL) write_flash $(flashimageoptions) $(INIT_BIN_ADDR) $(SDK_BASE)/bin/esp_init_data_default.bin $(BLANK_BIN_ADDR) $(SDK_BASE)/bin/blank.bin $(RBOOT_SPIFFS_0) $(SMING_HOME)/compiler/data/blankfs.bin
 
 rebuild: clean all
 

